src/main/java/net/earthcomputer/clientcommands/command/CTickCommand.java

```diff
@@ -0,0 +1,210 @@
+            .then(literal("client")
+                .then(literal("tps")
+                    .executes(ctx -> getClientTps(ctx.getSource())))
+                .then(literal("mspt")
+                    .executes(ctx -> getClientMspt(ctx.getSource()))))
+            .then(literal("server")
+                .then(literal("tps")
+                    .executes(ctx -> getServerTps(ctx.getSource())))
+                .then(literal("mspt")
+                    .executes(ctx -> getServerMspt(ctx.getSource())))));
+    }
+
+    private static int getClientTps(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        TickMeasuringTask measurer = new TickMeasuringTask(true, false);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float tps = source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.client.tps.expectedTps", tps));
+        return (int) tps;
+    }
+
+    private static int getClientMspt(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        TickMeasuringTask measurer = new TickMeasuringTask(false, false);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float mspt = TimeUtil.MILLISECONDS_PER_SECOND / source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.client.mspt.expectedMspt", mspt));
+        return (int) mspt;
+    }
+
+    private static int getServerTps(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        boolean isIntegratedServer = source.getClient().hasSingleplayerServer();
+        TickMeasuringTask measurer = new TickMeasuringTask(true, !isIntegratedServer);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float tps = source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.server.tps.expectedTps", tps));
+        return (int) tps;
+    }
+
+    private static int getServerMspt(FabricClientCommandSource source) throws CommandSyntaxException {
+        stopPreviousTask();
+
+        boolean isIntegratedServer = source.getClient().hasSingleplayerServer();
+        TickMeasuringTask measurer = new TickMeasuringTask(false, !isIntegratedServer);
+        TaskManager.addTask(TASK_NAME, measurer);
+        currentMeasurer = measurer;
+
+        float mspt = TimeUtil.MILLISECONDS_PER_SECOND / source.getWorld().tickRateManager().tickrate();
+        source.sendFeedback(Component.translatable("commands.ctick.server.mspt.expectedMspt", mspt));
+        return (int) mspt;
+    }
+
+    private static void stopPreviousTask() {
+        if (currentMeasurer != null) {
+            currentMeasurer._break();
+            TaskManager.removeTask(TASK_NAME);
```

Comment:
This is not how tasks work. The string argument of `addTask` is a template which only forms part of the task's full name, which `addTask` returns. This therefore won't remove the task properly. You also don't need to call `_break` because `removeTask` does that for you.